### PR TITLE
Add DeathAdder Chroma support

### PR DESCRIPTION
**Pull request by kzemek**
I reverse-engineered DeathAdder Chroma protocol and implemented a driver module for it. Other than that, the biggest changes are

* the addition of "LED mode" parameter (allowing to switch LEDs between static, spectrum and breathing modes) that triggered changes in the library and client-side code
* the change from `-std=gnu99` to `-std=c11`. The only two features I use in the code are anonymous structs/unions and `static_assert`, but I don't think the change is a big deal.

The changes are also described in some detail in commit messages.

### CODE REVIEW

**Comment by mbuesch** [comment]
Thanks a lot for your pull request.
A few comments on it:

- pyrazer/main.py / listFromSupportedModes():
> + if mask | (1 << mode):
This should probably be   mask &
Also: You might use a tuple instead of the list in the for statement.

- Same in hw_deathadder_chroma.c:
>+ if (!(d->dimension_mask | (1 << dim)))

- Coding style: No code in switch statements after "case foo: ...", please. Use a newline:
(Example):
> + case DEATHADDER_CHROMA_LED_ID_LOGO: return &d->logo_led;

- Coding style: Please re-indent the source code to match the code style of all other files (tabs). You can use the 'indent' tool for this.
I understand that everybody has his favorite style, however it is extremely important to keep styles inside a project consistent, no matter what style it actually is.

- Coding style: Please declare variables at the start of the function before any code. This is our coding style convention.

- Coding style: No space in front of function args, please. (See hw_deathadder_chroma.c:743 and others)

- Coding style: Also no space in front of equal-characters in assignments, please.

In short: Please adhere to the existing style ;)

**Comment by mbuesch** [comment]
Note that the coding style of this project is that of the Linux Kernel.
See here for a description:
https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/CodingStyle

**Comment by kzemek** [comment]
Sorry for the code style issues; I hope I've fixed them all. I used clang-format with [Linux-Kernel-like style](http://clang.llvm.org/docs/ClangFormatStyleOptions.html#examples) and the code seems consistent with the rest of the project, but there may be some corner cases left.

> if mask | (1 << mode): This should probably be mask &

Ouch! Definitely.
